component/web/security/src/main/java/org/exoplatform/web/register/ExternalRegisterHandler.java

```diff
@@ -401,8 +402,16 @@ private String createUser(User user) throws Exception {
     boolean isAdministrator = CollectionUtils.isNotEmpty(memberships);
     if (!isAdministrator && securitySettingService.isRegistrationExternalUser()) {
       // Avoid incoherence by indicating an admin user As external
-      deleteFromInternalUsersGroup(login);
-      addToExternalUsersGroup(login);
+      begin(organizationService);
```

Comment:
IMO, it would be better call a transitive transaction to ensure that it englobes a valid transaction for IDM listeners as well which may need a transaction on a different Datasource.

```suggestion
      RequestLifecycle.begin(ExoCotnainerContext.getCurrent...());
```

Reply:
BTW, a transaction should already be started due to this flag: `org.exoplatform.web.register.ExternalRegisterHandler.getRequiresLifeCycle()` which returns `true`

Reply:
If the transaction isn't started, we just need IMO to make a fix in `org.exoplatform.web.WebAppController.service(HttpServletRequest, HttpServletResponse)` rather than this specific Web Handler only. WDYT ?

Reply:
In fact, I wish to force the organizationService and Picketlink to persist the modifications. 
And I see that, in wrapForAutomaticLogin we have a restartTransaction. 

So I have to recheck this part

Reply:
ok, just to ensure that as result the IDM connection is closed, you check that the IDM Datasource has 0 active connections using jconsole on Datasource MBean.

hint: the restartTransaction can not work as expected due to a potential falsy condition in `org.exoplatform.web.WebAppController.service` (the returned container of `ExoContainerContext.getCurrentContainer()` has to be of type `PortalContainer`, else the IDM transaction isn't started, to check by debug mode if needed)

![image](https://github.com/user-attachments/assets/8a414357-618d-4f1f-89ff-6f5f9650c2cd)